src/edit_mode/vi/mod.rs

```diff
@@ -17,15 +17,24 @@ enum Mode {
 /// This parses incomming input `Event`s like a Vi-Style editor
 pub struct Vi {
     partial: Option<String>,
-    keybindings: Keybindings,
+    insert_keybindings: Keybindings,
+    normal_keybindings: Keybindings,
     mode: Mode,
 }
 
+fn default_vi_insert_mode_keybindings() -> Keybindings {
+    default_emacs_keybindings()
+}
+
+fn default_vi_normal_mode_keybindings() -> Keybindings {
+    Keybindings::new()
+}
+
 impl Default for Vi {
     fn default() -> Self {
         Vi {
-            // FIXME: Setup proper keybinds
-            keybindings: default_emacs_keybindings(),
+            insert_keybindings: default_vi_insert_mode_keybindings(),
+            normal_keybindings: default_vi_normal_mode_keybindings(),
```

Comment:
I think we always had different kebinds for normal and insert mode. So, readded this. Using emacs binds in insert mode, not sure if this is the best way or of any alternatives.

Reply:
insert mode is just for vim. Emacs has its own mode - I wonder why it used to use that for vim

Reply:
I think this was something that was introduced in the massive refactor I did, and I missed this piece, I will extract the default vi map and get it here.